pkg/kubelet/kubelet_pods.go

```diff
@@ -553,7 +550,7 @@ func (kl *Kubelet) makeEnvironmentVariables(pod *v1.Pod, container *v1.Container
 	// To avoid this users can: (1) wait between starting a service and starting; or (2) detect
 	// missing service env var and exit and be restarted; or (3) use DNS instead of env vars
 	// and keep trying to resolve the DNS name of the service (recommended).
-	serviceEnv, err := kl.getServiceEnvVarMap(pod.Namespace)
+	serviceEnv, err := kl.getServiceEnvVarMap(pod.Namespace, *pod.Spec.EnableServiceLinks)
```

Comment:
@bradhoekstra @thockin 
the panic i'm seeing is because of this line.

is the pod spec guarantied to be non-nil at this point?

Reply:
I was under the impression [this change](https://github.com/kubernetes/kubernetes/pull/68754/files#diff-f4a6ea0883571048398880099571eea7R185) to defaults.go would guarantee that.

If not, I can copy the defaulting logic to this file.

Reply:
Thanks for debugging this @neolit123

Reply:
> Thanks for debugging this @neolit123

@bradhoekstra 
no problem!
let me know if you want me to submit the PR as i have it ready here.

> I was under the impression this change to defaults.go would guarantee that.

i think it only does that when generating a default pod spec.

here is a similar case for `Pod.Spec.SecurityContext`:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kuberuntime/kuberuntime_sandbox.go#L159